### PR TITLE
MISC improvements: fix typescript and improve docs

### PR DESCRIPTION
**Pull request by anxolin**
This PR has some MISC changes to prepare the SDK to be consumed as a NPM package. 

Includes:
- Improve the README: Texts and small changes in the example
- Change the namespace of the package (use @gnosis.pm)
- Fix typescript issues, clients of this SDK wouldn't find the sources
- Fix some error importing from JSON the contract addresses

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -16,20 +16,25 @@
...
     "start": "microbundle -f modern,esm,cjs watch",
     "lint": "eslint src",
     "format": "prettier --write \"src/**/*.+(ts|json)\"",
-    "test": "jest"
+    "test": "jest",
+    "prepare": "npm run build",
+    "prepublishOnly": "npm test && npm run lint",
+    "preversion": "npm run lint",
+    "version": "npm run format && git add -A src",
+    "postversion": "git push && git push --tags"
```
Comment (anxolin) [inline package.json:35]: Some scripts to make sure the NPM packages pass some lint/formatting bars

Comment (anxolin) [reply package.json:35]: Also, make sure the NPM package contains the `dist` directory with the updated build

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
-  "name": "cow-sdk",
-  "version": "0.0.1-rc.0",
+  "name": "@gnosis.pm/cow-sdk",
```
Comment (anxolin) [inline package.json:2]: change namespace

File: package.json [hunk 3]
```diff
@@ -16,20 +16,25 @@
...
-    "default": "./dist/cow-sdk.modern.js"
+    "require": "./dist/src/cow-sdk.js",
+    "default": "./dist/src/cow-sdk.modern.js"
   },
-  "main": "./dist/cow-sdk.js",
-  "module": "./dist/cow-sdk.module.js",
-  "unpkg": "./dist/cow-sdk.umd.js",
+  "main": "./dist/src/cow-sdk.js",
+  "module": "./dist/src/cow-sdk.module.js",
+  "unpkg": "./dist/src/cow-sdk.umd.js",
```
Comment (anxolin) [inline package.json:24]: Some fixes, the previous setup was not allowing clients of the SDK to find the sources and typescript definitions. Also related to the change in `types`

File: src/constants/index.ts [hunk 4]
```diff
@@ -1,6 +1,8 @@
-import { GPv2Settlement } from '@gnosis.pm/gp-v2-contracts/networks.json'
+import contractNetworks from '@gnosis.pm/gp-v2-contracts/networks.json'
 import { SupportedChainId as ChainId } from './chains'
 
+const { GPv2Settlement } = contractNetworks
```
Comment (anxolin) [inline src/constants/index.ts:4]: There was a nasty error derived by the way we imported the GPv2Settlement from the JSON

File: README.md [hunk 5]
```diff
@@ -6,15 +6,19 @@
 [![Styled With Prettier](https://img.shields.io/badge/code_style-prettier-ff69b4.svg)](https://prettier.io/)
 
 
+> ⚠️⚠️ THE SDK IS IN Beta ⚠️⚠️
```
Comment (anxolin) [inline README.md:9]: disclaimer

**Comment by anxolin** [comment]
I need to merge this asap, pls do a post-merge review

File: package.json [hunk 6]
```diff
@@ -16,20 +16,25 @@
...
     "build": "microbundle -f modern,esm,cjs",
     "start": "microbundle -f modern,esm,cjs watch",
     "lint": "eslint src",
     "format": "prettier --write \"src/**/*.+(ts|json)\"",
-    "test": "jest"
+    "test": "jest",
+    "prepare": "npm run build",
+    "prepublishOnly": "npm test && npm run lint",
+    "preversion": "npm run lint",
+    "version": "npm run format && git add -A src",
```
Comment (alfetopito) [inline package.json:34]: These commands are a bit confusing to me.
What would be the workflow where you'd use each?

File: src/constants/index.ts (hunk 4 above)
Comment (alfetopito) [reply src/constants/index.ts:4]: Isn't this essentially the same thing as before?

Comment (anxolin) [reply src/constants/index.ts:4]: apparently not. If you try the previous you get an error when you try to use it as a library. 
tbh not sure what is the issue, but by reading about it I saw the official answer is that we should import it like this. It did the trick so i changed it.

File: package.json (hunk 6 above)
Comment (anxolin) [reply package.json:34]: they all run automatically when you publish a anew version

I copied the from somewhere, the most important above is the `prepare`, cause we were not building when publishing. The other are nice to have. 

* `prepare`: Runs any time before the package is packed, i.e. during npm publish and npm pack
* `prepublishOnly`: Runs BEFORE the package is prepared and packed, ONLY on npm publish.

The other doesn't harm, I'm fine with removing them. Although they make it easier to generate a version, cause it pushes the tags already. I got it here if it helps https://itnext.io/step-by-step-building-and-publishing-an-npm-typescript-package-44fe7164964c

Comment (alfetopito) [reply package.json:34]: Ah ok, they "magically" plug into something else.
Not against having them, just trying to understand how/where they were used

**Pull request merged**